ddk-parent/pom.xml

```diff
@@ -34,6 +34,7 @@
     <compiler.showWarnings>true</compiler.showWarnings>
     <pmd.ruleset>${workspace}/ddk-configuration/pmd/ruleset.xml</pmd.ruleset>
     <pmd.test.ruleset>${workspace}/ddk-configuration/pmd/ruleset-test.xml</pmd.test.ruleset>
+    <pmd.cpd.min>100000</pmd.cpd.min>

```

Comment:
I would add a comment why we are setting this that high and when it will be lowered again to what value (which is sensible?)

Reply:
This is set to such a high number to "ignore" duplicate code in the scan from parent pom. When scanning the child plugins we can optionally set this to a sensible value. The exclude root value doesn't actually work when checking from parent pom so we evaluate generated code and fail...

Reply:
I guess we should get rid of scanning on the parent and I have a strong guess that I introduced this with PMDs aggregate option which in this case has less benefits than drawbacks. Do you like to try to remove it and check whether this leads to better results?

In general the parent should just keep the configuration for the modules and do not trigger anything by itself. Also we should avoid setting config that makes no sense for the child modules and would need to be overwritten on any child module.

Reply:
Ok, sounds good.  I guess that means I can also change back to using test preferences too.

Reply:
Let's leave this for now and we consider CPD later but focus on PMD and Findbugs first

Reply:
Do you perhaps know how to disable CPD only?  I'd rather do that than change the value here, but nothing worked for me...